R/html-tweak.R

```diff
@@ -129,6 +129,58 @@ tweak_footnotes <- function(html) {
   xml2::xml_remove(container)
 }
 
+tweak_sourcecode_divs <- function(html) {
+  divs <- xml2::xml_find_all(html, ".//div[contains(@class, 'sourceCode')]")
+  if (length(divs) == 0) {
+    return()
+  }
+  purrr::walk(divs, tweak_sourcecode_div)
+}
+
+tweak_sourcecode_div <- function(html) {
+  pre_class <- xml2::xml_attr(xml2::xml_find_first(html, "pre"), "class")
+  is_r <- grepl("downlit", pre_class) || grepl("sourceCode [rR]", pre_class)
+  if (!is_r) {
+    text <- xml2::xml_text(html)
+  } else {
+    copy <- copy_html(html)
+    # remove comments
+    xml2::xml_remove(xml2::xml_find_all(copy, "//span[@class='co']"))
```

Comment:
I don't think we want to remove regular comments, just output?

Reply:
Ok, I wasn't sure what made the most sense!

Reply:
Why wouldn't downlit be active?

Reply:
I mean that except for examples downlit highlights code without evaluating it (i.e. it gets articles after they have been rendered) so can't guess which comnents are output (well it could if it can access the knitr option for each chunk to know what output comments start with, which could however be "" so indistinguishable ). This is why a knitr hook for output code for vignettes, articles (maybe the README if pkgdown were to render it) sounds useful.